src/DayCounts.jl

```diff
@@ -113,35 +113,36 @@ This ensures that:
  - [EMU and Market Conventions: Recent Developments, ISDA - BS:9951.1](https://www.isda.org/a/AIJEE/1998-ISDA-memo-EMU-and-Market-Conventions-Recent-Developments.pdf), §4. The Actual/Actual Day Count Convention.
 """
 struct ActualActualICMA <: DayCount
-    schedule::StepRange{Date,Month}
+    frequency
+    schedule::Vector{Date}
 end
+ActualActualICMA(schedule::StepRange{Date,Month}) = ActualActualICMA(Month(12)/step(schedule),collect(schedule))
 const ActualActualISMA = ActualActualICMA
 const ISMA99 = ActualActualICMA
 
 function yearfrac(startdate::Date, enddate::Date, dc::ActualActualICMA)
-    frequency = Month(12)/step(dc.schedule)
-
-    r1 = searchsorted(dc.schedule, startdate)
-    i1, j1 = last(r1), first(r1)
-    r2 = searchsorted(dc.schedule, enddate)
-    i2, j2 = last(r2), first(r2)
-
-    if i1 == i2 || j1 == j2
-        i = min(i1,i2)
-        j = max(j1,j2)
-        if i == j
-            return 0.0
-        else
-            return Dates.value(enddate - startdate) / (frequency * Dates.value(dc.schedule[j] - dc.schedule[i]))
-        end
-    end
+    if startdate > enddate
+        return -yearfrac(enddate,startdate,dc)
+    end      
+    startdate < first(dc.schedule) || enddate > last(dc.schedule) && error("The start and end dates must be contained within the scheduled dates.")
 
-    f1 = i1 == j1 ? 0.0 : (dc.schedule[j1] - startdate) / (dc.schedule[j1] - dc.schedule[i1])
-    f2 = i2 == j2 ? 0.0 : (enddate - dc.schedule[i2]) / (dc.schedule[j2] - dc.schedule[i2])
+    schedule = dc.schedule
 
-    (f1+f2+(i2-j1))/frequency
-end
+    periodfrac = 0.
+
+    # Fraction of the last period
+    istart = findlast(x -> x < enddate, schedule)
```

Comment:
`searchsorted` should be more efficient than `findlast` (since it can use binary rather than sequential search).

Reply:
Ok. I am not familiar with how `searchsorted` works, so maybe we merge this and submit a PR with `searchsorted` (and comments) later?